File: src/main/kotlin/com/api/socialNetwork/mapper/UserResponseMapper.java
package com.api.socialNetwork.mapper;

import java.util.Objects;

import org.springframework.stereotype.Component;

import com.api.socialNetwork.dtos.response.UserAccountResponse;
import com.api.socialNetwork.dtos.response.UserWithPostsResponse;
import com.api.socialNetwork.model.UserAccount;

@Component
public class UserResponseMapper {
	public UserAccountResponse toResponse (UserAccount user) {

		return new UserAccountResponse(
				Objects.requireNonNull(user.getUserId()),
				user.getUserName(),
				user.getNickname(),
				Objects.requireNonNull(user.getProfileImg()),
				user.getEmail()
 		);
	}

	public UserWithPostsResponse toResponseWithPosts (UserAccount user) {

		return new UserWithPostsResponse(
				toResponse(user),
				null
		);
	}
}

File: src/main/kotlin/com/api/socialNetwork/mapper/UserResponseMapper.kt
package com.api.socialNetwork.mapper

import com.api.socialNetwork.controller.dtos.response.UserAccountResponse
import com.api.socialNetwork.controller.dtos.response.UserWithPostsResponse
import com.api.socialNetwork.model.UserAccount
import org.springframework.stereotype.Component

@Component
class UserResponseMapper {
    fun toResponse(user: UserAccount): UserAccountResponse {
        return UserAccountResponse(
            user.userId!!,
            user.username,
            user.nickname,
            user.profileImg,
            user.email
        )
    }

    fun toResponse(userList: List <UserAccount>): List<UserAccountResponse> {
       return userList.map {userAccount -> toResponse(userAccount) }
    }

    fun toResponseWithPosts(user: UserAccount): UserWithPostsResponse {
        return UserWithPostsResponse(
            toResponse(user),
            null
        )
    }

}
